invenio_rdm_records/services/services.py

```diff
@@ -516,6 +535,22 @@ def read(self, identity, id_, expand=False, include_deleted=False):
 
         return result
 
+    def read_draft(self, identity, id_, expand=False):
+        """Retrieve a draft of a record.
+
+        If the draft has a "deleted" published record then we return 410.
+        """
+        result = super().read_draft(identity, id_, expand=expand)
+        # check that if there is a published deleted record then return 410
+        draft = result._record
+        if draft.is_published:
+            record = self.record_cls.pid.resolve(id_)
+            if record.deletion_status.is_deleted:
+                result = super().read(identity, id_, expand=expand)
+                raise RecordDeletedException(record, result_item=result)
```

Comment:
this will result in `/uploads/<id>` giving back the record's tombstone's page

![Screenshot 2023-10-23 at 15 24 25](https://github.com/inveniosoftware/invenio-rdm-records/assets/22594783/393f4ef2-8060-47eb-bd1b-c23d2159e1d5)